### PR TITLE
chore: use chain aware address

### PR DESCRIPTION
**Pull request by 0xonramp**
Use `getAddressForToken` to get chain-aware address where possible.

### CODE REVIEW

File: src/lib/hooks/use-best-quote/utils/issuance/issuance-quote.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ interface IssuanceQuoteRequest {
   slippage: number
 }
 
+// In scope?
```
Comment (0xonramp) [inline src/lib/hooks/use-best-quote/utils/issuance/issuance-quote.ts:30]: @janndriessen this file along with `redemption-quote.ts` both are hardcoded to use `chainId: 1`, are there plans to eventually support other chains here, or is it out of scope?

File: src/lib/utils/api/coingecko.ts [hunk 2]
```diff
@@ -52,20 +51,3 @@ export const fetchCoingeckoTokenPrice = async (
...
   return data[address.toLowerCase()][baseCurrency]
 }
-
-type CoingeckoMarketData = {
-  current_price: { usd: number }
-  market_cap: { usd: number }
-  price_change_percentage_24h_in_currency: { usd: number }
-}
-
-export const fetchCoingeckoMarketData = async (
```
Comment (0xonramp) [inline src/lib/utils/api/coingecko.ts:62]: Unused in the codebase.

File: src/lib/hooks/use-best-quote/utils/issuance/issuance-quote.ts (hunk 1 above)
Comment (janndriessen) [reply src/lib/hooks/use-best-quote/utils/issuance/issuance-quote.ts:30]: I'd leave these out for now - as I don't think they will change anytime soon. But reminded me to add this ticket: https://github.com/IndexCoop/index-app/issues/1321. Additionally, we could think about moving these to the `/quote` endpoint as well some time?

File: src/lib/hooks/use-best-quote/utils/index-quote.ts [hunk 3]
```diff
@@ -59,7 +60,9 @@ export async function getIndexQuote(
   } = request
   try {
     const inputAmount = parseUnits(inputTokenAmount, inputToken.decimals)
-    const path = `/quote?takerAddress=${address}&inputToken=${inputToken.address}&outputToken=${outputToken.address}&inputAmount=${inputAmount.toString()}&chainId=${chainId}`
+    const inputTokenAddress = getAddressForToken(inputToken, chainId)
+    const outputTokenAddress = getAddressForToken(outputToken, chainId)
+    const path = `/quote?takerAddress=${address}&inputToken=${inputTokenAddress}&outputToken=${outputTokenAddress}&inputAmount=${inputAmount.toString()}&chainId=${chainId}`
```
Comment (janndriessen) [inline src/lib/hooks/use-best-quote/utils/index-quote.ts:65]: nice, making it future proof. 🌟

File: src/lib/hooks/use-best-quote/utils/issuance/redemption-quote.ts [hunk 4]
```diff
@@ -28,6 +28,7 @@ interface RedemptionQuoteRequest {
 
 const contract = DebtIssuanceModuleAddress
 
+// In scope?
```
Comment (janndriessen) [inline src/lib/hooks/use-best-quote/utils/issuance/redemption-quote.ts:31]: see above

**Review by janndriessen — APPROVED** [review]

**Pull request merged**